libraries/AP_Scripting/lua_bindings.cpp

```diff
@@ -250,11 +250,111 @@ const luaL_Reg AP_Logger_functions[] = {
     {NULL, NULL}
 };
 
+static int get_analog_in(lua_State *L) {
+
+    AP_HAL::AnalogSource *data = hal.analogin->channel(ANALOG_INPUT_NONE);
```

Comment:
This either needs the global lock, or you need to manage some locking inside AP_HAL level. (The latter is actually a lot better, but this method allocates, and it's not externally obvious how this will break stuff from outside of looking in the implementation).

Reply:
a note for future me, fix it here: https://github.com/ArduPilot/ardupilot/blob/d6dc8864db582951634586978d0275c34a23ca87/libraries/AP_HAL_ChibiOS/AnalogIn.cpp#L346

Reply:
Added a semaphore

Reply:
You need to look into HAL_Linux analog in as well. Looks like HAL_SITL is fine, but HAL_Linux needs the same protections.

Reply:
your right, I think I got them all now.